app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedFragmentPresenter.kt

```diff
@@ -0,0 +1,195 @@
+  ) {
+    binding.viewModel = model
+    if (model.checkIfChapterIsCompleted()) {
+      binding.isChapterChecked = true
+      binding.isChapterCheckboxEnabled = false
+    } else {
+      binding.isChapterChecked =
+        selectedExplorationIdList.contains(model.chapterSummary.explorationId)
+
+      binding.isChapterCheckboxEnabled = !model.chapterSummary.hasMissingPrerequisiteChapter() ||
+        model.chapterSummary.hasMissingPrerequisiteChapter() &&
+        selectedExplorationIdList.contains(
+          model.chapterSummary.missingPrerequisiteChapter.explorationId
+        )
+
+      binding.markChaptersCompletedChapterCheckBox.setOnCheckedChangeListener { _, isChecked ->
+        if (isChecked) {
+          chapterSelected(
+            model.chapterIndex,
+            model.nextStoryIndex,
+            model.chapterSummary.explorationId
+          )
+        } else {
+          chapterUnselected(model.chapterIndex, model.nextStoryIndex)
+        }
+      }
+    }
+  }
+
+  private fun getMarkChaptersCompletedViewModel(): MarkChaptersCompletedViewModel {
+    return viewModelProvider.getForFragment(fragment, MarkChaptersCompletedViewModel::class.java)
+  }
+
+  override fun chapterSelected(chapterIndex: Int, nextStoryIndex: Int, explorationId: String) {
+    if (!selectedExplorationIdList.contains(explorationId)) {
+      selectedExplorationIdList.add(explorationId)
+    }
+    if (selectedExplorationIdList.size ==
+      getMarkChaptersCompletedViewModel().getItemList().count {
+        it is ChapterSummaryViewModel && !it.checkIfChapterIsCompleted()
+      }
+    ) {
+      binding.isAllChecked = true
+    }
+    if (!binding.markChaptersCompletedRecyclerView.isComputingLayout &&
+      binding.markChaptersCompletedRecyclerView.scrollState == RecyclerView.SCROLL_STATE_IDLE
+    ) {
+      bindingAdapter.notifyItemChanged(chapterIndex)
+      if (chapterIndex + 1 < nextStoryIndex)
+        bindingAdapter.notifyItemChanged(chapterIndex + 1)
+    }
+  }
+
+  override fun chapterUnselected(chapterIndex: Int, nextStoryIndex: Int) {
+    for (index in chapterIndex until nextStoryIndex) {
+      val explorationId =
+        (getMarkChaptersCompletedViewModel().getItemList()[index] as ChapterSummaryViewModel)
+          .chapterSummary.explorationId
+      if (selectedExplorationIdList.contains(explorationId)) {
+        selectedExplorationIdList.remove(explorationId)
+      }
+    }
+    if (selectedExplorationIdList.size !=
+      getMarkChaptersCompletedViewModel().getItemList().count {
+        it is ChapterSummaryViewModel && !it.checkIfChapterIsCompleted()
+      }
+    ) {
+      binding.isAllChecked = false
+    }
+    if (!binding.markChaptersCompletedRecyclerView.isComputingLayout &&
+      binding.markChaptersCompletedRecyclerView.scrollState == RecyclerView.SCROLL_STATE_IDLE
+    ) {
+      bindingAdapter.notifyItemRangeChanged(
+        chapterIndex,
+        /*itemCount = */nextStoryIndex - chapterIndex
```

Comment:
Nit space `/*itemCount = */ nextStoryIndex - chapterIndex`

Reply:
Done. Also, thanks for the review.